src/Toast.js

```diff
@@ -73,6 +79,17 @@ const Toast = React.forwardRef(
     bsPrefix = useBootstrapPrefix('toast');
     const delayRef = useRef(delay);
     const onCloseRef = useRef(onClose);
+    const [exited, setExited] = useState(!show);
+
+    const useAnimation = Transition && animation;
+
+    if (show) {
+      if (exited) {
+        setExited(false);
+      }
+    } else if (!useAnimation && !exited) {
+      setExited(true);
+    }
```

Comment:
This should be in a `useEffect` call, and it should also be in a functional update.
```suggestion
useEffect(() => {
  setExited(prevExited => {
    if (show && prevExited) {
      return false;
    } else if (!useAnimation && !prevExited) {
      return true;
    } else {
      return prevExited;
    }
  });
}, [show, useAnimation]);
```